### ISSUE TITLE
[v2 BUG]: The extra metrics calcualtions don't work for `MveFFN` and `EvidentialFFN`

### ISSUE THREAD

**Issue by shihchengli** [chemprop/chemprop#944]
**Describe the bug**
For metrics calculations, we currently use all the model outputs in the metrics function, which causes issues for the `MveFFN` and `EvidentialFFN` models. For example, if RMSE is to be calculated, only the predicted property should be used as input in the metrics, but currently, all the other parameters (variance for MVE; $\lambda$, $\alpha$, and $\beta$ for the evidential model) are used as inputs, causing the shape of Tensors for predictions and targets inconsistent during metrics calculations.

Since every loss function can be treated as a metric, we sometimes need the other parameters to be included in the metrics. I noticed this kind of metrics includes `MVELoss` and `EvidentialLoss` (also quantile loss, which will be implemented in v2.1), but they were not defined in the metrics.py, so I think we don't really treat them as metrics that we want to report to measure the model performance. If we think those loss functions should be avoided as metrics and instead used as criteria to guide model optimization, we should be able to just use the predicted property values from the model outputs as input to the metrics. In this case, we can use `self.criterion` to calculate `val_loss` and remove it from the `self.metrics` defined in the MPNN object (see [here](https://github.com/chemprop/chemprop/blob/dae03434253417112a29b9a1f502caf731c78450/chemprop/models/model.py#L91-L95)). Some discussions related to this topic can also be found in #817.

**Checklist**
- [ ] all dependencies are satisifed: `conda list` or `pip list` shows the packages listed in the `pyproject.toml`
- [ ] the unit tests are working: `pytest -v` reports no errors
